File: src/contests/cg/puzzle/clojure/the-river-i-.clj
;; 🎮 CodinGame Puzzle - the-river-i-
;; https://www.codingame.com/training/easy/the-river-i-

(ns Solution (:gen-class))

(defn digit-sum [x]
  (loop [x x s 0]
    (if (> x 0) (recur (quot x 10) (+ s (rem x 10))) s)))

(defn -main [& _]
  (let [a (Long/parseLong (.trim (read-line)))
        b (Long/parseLong (.trim (read-line)))]
    (loop [a a b b]
      (cond
        (= a b) (println a)
        (< a b) (recur (+ a (digit-sum a)) b)
        :else (recur a (+ b (digit-sum b))))))
  (flush))

File: src/contests/cg/puzzle/lua/the-river-i-.lua
-- 🎮 CodinGame Puzzle - the-river-i-
-- https://www.codingame.com/training/easy/the-river-i-

local function digit_sum(x)
  local s = 0
  while x > 0 do
    s = s + x % 10
    x = x // 10
  end
  return s
end

local a = tonumber(io.read())
local b = tonumber(io.read())
while a ~= b do
  if a < b then
    a = a + digit_sum(a)
  else
    b = b + digit_sum(b)
  end
end
print(a)
